nima/http/processor/src/main/java/io/helidon/nima/http/processor/HttpMethodProducer.java

```diff
@@ -0,0 +1,393 @@
+        return tools.produceStandardCodeGenResponse(request,
+                                                    generatedType,
+                                                    () -> Templates.loadTemplate("nima", "http-method.java.hbs"),
+                                                    it -> addProperties(request, it));
+    }
+
+    private Map<String, Object> addProperties(CustomAnnotationTemplateRequest request,
+                                              Map<String, Object> currentProperties) {
+        TypedElementName targetElement = request.targetElement();
+        Map<String, Object> response = new HashMap<>(currentProperties);
+
+        HttpDef http = new HttpDef();
+        /*
+           Method response
+           http.response.type
+           http.response.isVoid
+         */
+        TypeName returnType = targetElement.typeName();
+        if ("void".equals(returnType.className())) {
+            http.response = new HttpResponse();
+        } else {
+            http.response = new HttpResponse(returnType.name());
+        }
+
+        // http.methodName - name of the method in source code (not HTTP Method)
+        http.methodName = targetElement.elementName();
+
+        // http.params (full string)
+        List<HeaderDef> headerList = new LinkedList<>();
+        List<TypedElementName> elementArgs = request.targetElementArgs();
+        LinkedList<String> parameters = new LinkedList<>();
+        int headerCount = 1;
+        for (TypedElementName elementArg : elementArgs) {
+            String type = elementArg.typeName().name();
+
+            switch (type) {
+            case "io.helidon.nima.webserver.http.ServerRequest" -> parameters.add("req,");
+            case "io.helidon.nima.webserver.http.ServerResponse" -> parameters.add("res,");
+            default -> processParameter(http, parameters, headerList, type, elementArg);
+            }
+        }
+
+        if (!parameters.isEmpty()) {
+            String last = parameters.removeLast();
+            last = last.substring(0, last.length() - 1);
+            parameters.addLast(last);
+        }
+        http.params = parameters;
+
+        /*
+            Headers
+            http.headers, field, name
+         */
+        http.headers = headerList;
+
+        /*
+            HTTP Method
+            http.method
+         */
+        if (request.annoTypeName().className().equals(HTTP_METHOD_ANNOTATION)) {
+            http.method = findAnnotValue(targetElement.annotations(), HTTP_METHOD_ANNOTATION, null);
+            if (http.method == null) {
+                throw new IllegalStateException("HTTP method producer called without HTTP Method annotation (such as @GET)");
+            }
+        } else {
+            http.method = request.annoTypeName().className();
+        }
+
+        // HTTP Path (if defined)
+        http.path = findAnnotValue(targetElement.annotations(), PATH_ANNOTATION, "");
+
+        response.put("http", http);
```

Comment:
constant def needed for "http"

Reply:
why? It is the only place it is used - this is only used from the template expression.